### PR TITLE
Reduce local services to only those in use

### PR DESCRIPTION
**Pull request by h-lame**
For: https://trello.com/c/bXJG1RU8/401-investigate-and-fix-data-discrepancies-3

We want to use the local_services.csv as an indicator of which LGSL codes are actually used by GOV.UK documents.  To this end we remove all rows from it that are not currently used by a local transaction and provide rake tasks to clean up the now "orphaned" instances that would be left in the DB.

There's also a db migration which should be run before the rake task to fix the LGSL code used in an archived edition.

See the individual commits for a more verbose description of why we're doing this.

### CODE REVIEW

File: test/unit/lib/local_service_cleaner_test.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require_relative '../../test_helper'
+
+class LocalServiceCleanerTest < ActiveSupport::TestCase
+
+  context "removing services we don't need anymore" do
+    setup do
+      data = "LGSL,Description,Providing Tier\n"
+      data += "1234,Finding Joy,all\n"
+      data += "5678,Escaping Sadness,county/unitary\n"
```
Comment (issyl0) [inline test/unit/lib/local_service_cleaner_test.rb]: 👏

File: lib/local_service_cleaner.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+require 'set'
+
+class LocalServiceCleaner
+  def initialize(input_data = LocalServiceImporter.fetch_data)
+    @input_data = input_data
+  end
+
+  DetectionResult = Struct.new(:local_service, :"in_csv?", :"in_local_transaction?")
```
Comment (issyl0) [inline lib/local_service_cleaner.rb]: What does this do? 💭

Comment (h-lame) [inline lib/local_service_cleaner.rb]: Prove that I'm a messy developer who doesn't tidy up vestigial tails of alternative versions I decided against showing to the world.  Will delete.

File: test/unit/lib/local_service_cleaner_test.rb (hunk 1 above)
Comment (jennyd) [inline test/unit/lib/local_service_cleaner_test.rb]: Districts can't escape sadness? 😢

Comment (h-lame) [inline test/unit/lib/local_service_cleaner_test.rb]: Handled at the county level, innit?  More efficient at that scale.

File: lib/tasks/local_transactions.rake [hunk 3]
```diff
@@ -9,6 +9,7 @@ namespace :local_transactions do
   task :fetch_and_clean => :environment do
     Rake::Task["local_transactions:fetch"].invoke
     Rake::Task["check_for_ghosts:remove"].invoke
+    Rake::Task["local_transactions:remove_old_services"]
```
Comment (davidbasalla) [inline lib/tasks/local_transactions.rake]: Should this be `Rake::Task["local_transactions:remove_old_services"].invoke` ?

Comment (h-lame) [inline lib/tasks/local_transactions.rake]: ugh, yes.  Good spot!

Comment (h-lame) [inline lib/tasks/local_transactions.rake]: Resolved via rebase.  Sorry!

**Pull request merged**